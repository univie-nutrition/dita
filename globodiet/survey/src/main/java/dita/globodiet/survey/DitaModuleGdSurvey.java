/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 *       http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
// Auto-generated by Causeway-Stuff code generator.
package dita.globodiet.survey;

import dita.globodiet.survey.dom.Campaign;
import dita.globodiet.survey.dom.CampaignDeps;
import dita.globodiet.survey.dom.Campaign_delete;
import dita.globodiet.survey.dom.Campaign_survey;
import dita.globodiet.survey.dom.Survey;
import dita.globodiet.survey.dom.SurveyDeps;
import dita.globodiet.survey.dom.Survey_delete;
import java.lang.Class;
import java.lang.String;
import javax.annotation.processing.Generated;
import org.apache.causeway.commons.collections.Can;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

@Generated("io.github.causewaystuff.companion.codegen.domgen._GenModule")
@Configuration
@Import({

        // Config Beans
        ModuleConfig.class,

        // Menu Entries
        EntitiesMenu.class,

        // Entities
        Campaign.class,
        Survey.class,

        // Submodules
        CampaignDeps.class,
        SurveyDeps.class,

        // Mixins
        Campaign_delete.class,
        Campaign_survey.class,
        Survey_delete.class,
        })
public class DitaModuleGdSurvey {
    public static final String NAMESPACE = "dita.globodiet.survey";

    public static Can<Class<?>> entityClasses() {
        return Can.of(Campaign.class,
        Survey.class);
    }
}
