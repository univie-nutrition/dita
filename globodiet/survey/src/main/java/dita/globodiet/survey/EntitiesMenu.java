/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 *       http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
// Auto-generated by Causeway-Stuff code generator.
package dita.globodiet.survey;

import dita.globodiet.survey.dom.Campaign;
import dita.globodiet.survey.dom.ReportColumnDefinition;
import dita.globodiet.survey.dom.RespondentFilter;
import dita.globodiet.survey.dom.Survey;
import io.github.causewaystuff.companion.applib.services.search.SearchService;
import jakarta.inject.Inject;
import jakarta.inject.Named;
import javax.annotation.processing.Generated;
import org.apache.causeway.applib.annotation.Action;
import org.apache.causeway.applib.annotation.ActionLayout;
import org.apache.causeway.applib.annotation.DomainService;
import org.apache.causeway.applib.services.factory.FactoryService;

@Generated("io.github.causewaystuff.companion.codegen.domgen._GenMenu")
@Named("dita.globodiet.survey.EntitiesMenu")
@DomainService
public class EntitiesMenu {
    @Inject
    private FactoryService factoryService;

    @Inject
    private SearchService searchService;

    @Action
    @ActionLayout(
            cssClassFa = "solid users-viewfinder .campaign-color"
    )
    public Campaign.Manager manageCampaign() {
        return factoryService.viewModel(new Campaign.Manager(searchService, ""));
    }

    @Action
    @ActionLayout(
            cssClassFa = "solid table-columns .reportColumnDefinition-color\n"
    )
    public ReportColumnDefinition.Manager manageReportColumnDefinition() {
        return factoryService.viewModel(new ReportColumnDefinition.Manager(searchService, ""));
    }

    @Action
    @ActionLayout(
            cssClassFa = "solid user .respondentFilter-color,\n"
                            + "solid filter .survey-color .ov-size-60 .ov-right-55 .ov-bottom-55\n"
    )
    public RespondentFilter.Manager manageRespondentFilter() {
        return factoryService.viewModel(new RespondentFilter.Manager(searchService, ""));
    }

    @Action
    @ActionLayout(
            cssClassFa = "solid users .survey-color"
    )
    public Survey.Manager manageSurvey() {
        return factoryService.viewModel(new Survey.Manager(searchService, ""));
    }
}
