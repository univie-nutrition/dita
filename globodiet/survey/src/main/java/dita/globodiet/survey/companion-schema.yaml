# -----------------------------------------------------------
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
# 
#       http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
# -----------------------------------------------------------

module:
  namespace: dita.globodiet.survey
  package: dita.globodiet.survey
viewmodels:
entities:
- id: dom.Campaign
  name: Campaign
  namespace: dom
  table: CAMPAIGN
  secondaryKey: |
    SURVEY
    CODE
  title: String.format("%s (code=%s|%s)", name, surveyCode, code)
  icon: solid users-viewfinder .campaign-color
  description: |
    A campaign defines a part of a food consumption survey that contains several interviews.
    Campaigns can be defined to be the whole study, 
    a seasonal part of a study, a regional part of a study etc.
  fields:
    surveyCode:
      column: SURVEY
      column-type: nvarchar(20)
      required: true
      unique: false
      foreignKeys: |
        SURVEY.CODE
      description: |
        Survey code   
    code:
      column: CODE
      column-type: nvarchar(20)
      required: true
      unique: false
      description: |
        Unique (survey scoped) campaign identifier.
    name:
      column: NAME
      column-type: nvarchar(120)
      required: true
      unique: false
      description: |
        Descriptive campaign name.
    description:
      column: DESCRIPTION
      column-type: nvarchar(240)
      required: false
      unique: false
      multiLine: 4
      description: |
        Detailed information for this campaign.
- id: dom.ReportColumnDefinition
  name: ReportColumnDefinition
  namespace: dom
  table: ReportColumnDefinition
  secondaryKey: |
    SURVEY
    NAME
  title: String.format("%s (%s)", name, surveyCode)
  icon: |
    solid table-columns .reportColumnDefinition-color
    
  description: |
    A column-definition defines which columns to include 
    with an interview report.
  fields:
    surveyCode:
      column: SURVEY
      column-type: nvarchar(20)
      required: true
      unique: false
      foreignKeys: |
        SURVEY.CODE
      description: |
        Survey code   
    name:
      column: NAME
      column-type: nvarchar(120)
      required: true
      unique: false
      description: |
        Descriptive column-definition name, unique within survey.
    description:
      column: DESCRIPTION
      column-type: nvarchar(240)
      required: false
      unique: false
      multiLine: 4
      description: |
        Detailed information for this column-definition.
    columnListing:
      column: COLLISTING
      column-type: mediumtext
      required: false
      unique: false
      cssClass: listing
      fieldSet: listing
      multiLine: 24
      hiddenWhere: EVERYWHERE
      labelPosition: NONE
      description: |
        Line by line defines a column to include.
        (lines can be commented out with a leading #)
- id: dom.RespondentFilter
  name: RespondentFilter
  namespace: dom
  table: RespondentFilter
  secondaryKey: |
    SURVEY
    NAME
  title: String.format("%s (%s)", name, surveyCode)
  icon: |
    solid user .respondentFilter-color,
    solid filter .survey-color .ov-size-60 .ov-right-55 .ov-bottom-55
    
  description: |
    A respondent filter defines which respondents to include 
    with an interview report.
  fields:
    surveyCode:
      column: SURVEY
      column-type: nvarchar(20)
      required: true
      unique: false
      foreignKeys: |
        SURVEY.CODE
      description: |
        Survey code   
    name:
      column: NAME
      column-type: nvarchar(120)
      required: true
      unique: false
      description: |
        Descriptive filter name, unique within survey.
    description:
      column: DESCRIPTION
      column-type: nvarchar(240)
      required: false
      unique: false
      multiLine: 4
      description: |
        Detailed information for this filter.
    aliasListing:
      column: ALIAS
      column-type: mediumtext
      required: false
      unique: false
      cssClass: listing
      fieldSet: listing
      multiLine: 24
      hiddenWhere: EVERYWHERE
      labelPosition: NONE
      description: |
        Line by line defines an alias to include.
        (lines can be commented out with a leading #)
- id: dom.Survey
  name: Survey
  namespace: dom
  table: SURVEY
  secondaryKey: |
    CODE
  title: String.format("%s (code=%s)", name, code)
  icon: solid users .survey-color
  description: |
    A survey collects one or more campaigns.
  fields:
    code:
      column: CODE
      column-type: nvarchar(20)
      required: true
      unique: true
      description: |
        Unique (application scoped) survey identifier.
    name:
      column: NAME
      column-type: nvarchar(120)
      required: true
      unique: false
      description: |
        Descriptive survey name.
    systemId:
      column: SYSID
      column-type: nvarchar(20)
      required: true
      unique: false
      description: |
        System ID part of semantic identifiers for this survey.
        e.g. at.gd/2.0
    correction:
      column: CORRECTION
      column-type: clob
      required: false
      unique: false
      hiddenWhere: EVERYWHERE
      description: |
        Yaml formatted interview data corrections.

