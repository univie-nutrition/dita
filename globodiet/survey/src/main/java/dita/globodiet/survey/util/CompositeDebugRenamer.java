/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  or more contributor license agreements.  See the NOTICE file
 *  distributed with this work for additional information
 *  regarding copyright ownership.  The ASF licenses this file
 *  to you under the Apache License, Version 2.0 (the
 *  "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 */
package dita.globodiet.survey.util;

import java.util.concurrent.atomic.LongAdder;

import dita.recall24.dto.RecallNode24;
import dita.recall24.dto.RecallNode24.Transfomer;
import dita.recall24.dto.Record24;
import dita.recall24.dto.Record24.Composite;

/**
 * Adds a composite number of occurrence.
 *
 * <p>Phase 2 processes all the candidates as collected from phase 1.
 */
public record CompositeDebugRenamer() implements Transfomer {

    private static final LongAdder c = new LongAdder();

    @SuppressWarnings("unchecked")
    @Override
    public <T extends RecallNode24> T transform(final T node) {
        return switch (node) {
            case Record24.Composite composite -> {
                c.increment();
                var builder = ((Composite.Builder) composite.asBuilder())
                        .name(composite.name() + "(" + c.longValue() + ")");
                builder.replaceSubRecords(this::transform); // recursive
                yield (T) builder.build();
            }
            default -> node;
        };
    }

}
