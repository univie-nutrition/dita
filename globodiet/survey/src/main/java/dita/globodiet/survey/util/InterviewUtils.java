/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  or more contributor license agreements.  See the NOTICE file
 *  distributed with this work for additional information
 *  regarding copyright ownership.  The ASF licenses this file
 *  to you under the Apache License, Version 2.0 (the
 *  "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 */
package dita.globodiet.survey.util;

import java.nio.charset.StandardCharsets;
import java.util.Optional;
import java.util.stream.Stream;

import org.apache.causeway.applib.value.Blob;
import org.apache.causeway.applib.value.Clob;
import org.apache.causeway.applib.value.NamedWithMimeType.CommonMimeType;
import org.apache.causeway.commons.io.ZipUtils.ZipOptions;

import lombok.experimental.UtilityClass;
import lombok.extern.log4j.Log4j2;

import io.github.causewaystuff.blobstore.applib.BlobDescriptor;
import io.github.causewaystuff.blobstore.applib.BlobStore;
import io.github.causewaystuff.commons.base.types.NamedPath;

@UtilityClass
@Log4j2
public class InterviewUtils {

    public Clob unzip(final Blob zippedInterviewSource) {
        var unzippedBlob = zippedInterviewSource
                .unZip(CommonMimeType.XML, ZipOptions.builder()
                        .zipEntryCharset(StandardCharsets.ISO_8859_1)
                        .zipEntryFilter(entry->{
                            log.info(String.format("parsing Zip entry %s (%.2fKB)",
                                    entry.getName(),
                                    0.001*entry.getSize()));
                            return true;
                        })
                        .build());
        return unzippedBlob
                .toClob(StandardCharsets.UTF_8);
    }

    public Stream<Clob> streamSources(final BlobStore surveyBlobStore, final NamedPath path, final boolean recursive) {
        return surveyBlobStore.listDescriptors(path, recursive)
            .stream()
            .filter(desc->desc.mimeType().equals(CommonMimeType.XML))
            .map(BlobDescriptor::path)
            .map(surveyBlobStore::lookupBlob)
            .map(Optional::orElseThrow)
            .map(InterviewUtils::unzip);
    }

}
