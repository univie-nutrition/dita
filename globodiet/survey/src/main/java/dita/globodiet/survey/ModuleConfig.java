/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
// Auto-generated by DitA-Tooling
package dita.globodiet.survey;

import org.causewaystuff.blobstore.applib.BlobStore;
import org.causewaystuff.blobstore.applib.BlobStoreFactory;
import org.causewaystuff.commons.base.types.NamedPath;

import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

import dita.globodiet.survey.entities.SurveyManager_addSurvey;
import dita.globodiet.survey.entities.Survey_addCampaign;
import dita.globodiet.survey.util.InterviewUtils;
import dita.globodiet.survey.view.SurveyTreeNode;
import dita.globodiet.survey.view.SurveyTreeNodeFactory;
import dita.globodiet.survey.view.SurveyVM;
import dita.recall24.model.InterviewSet24;

@Configuration
@Import({
    SurveyVM.class,
    BlobStoreFactory.class,
    
    // non-gen mixins
    SurveyManager_addSurvey.class,
    Survey_addCampaign.class,
        })
@EnableConfigurationProperties({ModuleConfig.SurveyConfiguration.class})
//@ComponentScan(basePackageClasses = Survey.class)
public class ModuleConfig {
    
    @ConfigurationProperties(DitaModuleGdSurvey.NAMESPACE)
    public static record SurveyConfiguration(
            BlobStoreFactory.BlobStoreConfiguration blobstore) {
    }

    @Bean @Qualifier("survey")
    public BlobStore surveyBlobStore(
            final BlobStoreFactory blobStoreFactory,
            final SurveyConfiguration surveyConfiguration) {
        return blobStoreFactory.createBlobStore(surveyConfiguration.blobstore());
    }

    @Bean
    public SurveyTreeNode surveyTreeRootNode(@Qualifier("survey") final BlobStore surveyBlobStore) {

        if(surveyBlobStore==null) {
            return SurveyTreeNodeFactory.emptyNode();
        }

        return InterviewUtils.streamSources(surveyBlobStore, NamedPath.empty(), true)
            .limit(1)
            .map(InterviewUtils::parse)
            .reduce(InterviewSet24::join)
            .map(InterviewSet24::normalized)
            .map(SurveyTreeNodeFactory::surveyNode)
            .orElseGet(SurveyTreeNodeFactory::emptyNode);
    }

}
