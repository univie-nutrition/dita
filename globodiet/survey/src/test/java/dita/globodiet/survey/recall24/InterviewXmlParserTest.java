/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  or more contributor license agreements.  See the NOTICE file
 *  distributed with this work for additional information
 *  regarding copyright ownership.  The ASF licenses this file
 *  to you under the Apache License, Version 2.0 (the
 *  "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 */
package dita.globodiet.survey.recall24;

import jakarta.inject.Inject;

import org.causewaystuff.blobstore.applib.BlobStore;
import org.causewaystuff.commons.types.NamedPath;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.boot.test.context.SpringBootTest;

import org.apache.causeway.commons.io.DataSource;

import dita.globodiet.survey.DitaModuleGdSurvey;
import dita.globodiet.survey.PrivateDataTest;
import dita.globodiet.survey.util.InterviewUtils;

@SpringBootTest(classes = {
        DitaModuleGdSurvey.class
        })
@PrivateDataTest
class InterviewXmlParserTest {

    @Inject @Qualifier("survey") BlobStore surveyBlobStore;
    private InterviewXmlParser parser = new InterviewXmlParser();

    @Test
    void parsingFromBlobStore() {
        assertNotNull(surveyBlobStore);

        InterviewUtils.streamSources(surveyBlobStore, NamedPath.empty(), true)
        .limit(1)
        .map(InterviewUtils::parse)
        .forEach(iSet->{
            System.err.printf("%s%n", iSet.toJson());
        });
    }

    @Test
    void parsingSample() {
        var xml = InterviewSampler.sampleXml();
        var iSet = parser.parse(DataSource.ofStringUtf8(xml));
        //System.err.printf("%s%n", iSet.toJson());
    }

}
