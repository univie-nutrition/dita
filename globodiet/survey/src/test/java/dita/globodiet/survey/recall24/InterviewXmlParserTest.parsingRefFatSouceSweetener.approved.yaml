respondents:
- alias: "1234"
  dateOfBirth: "2000-01-01"
  sex: "FEMALE"
  interviews:
  - interviewDate: "2024-05-28"
    interviewOrdinalRef:
      value: 1
    respondentSupplementaryData:
      specialDietId: "00"
      specialDayId: "00"
      heightCM: 160
      weightKG: 60
    meals:
    - hourOfDay: "09:00:00"
      foodConsumptionOccasionId: "02"
      foodConsumptionPlaceId: "01"
      memorizedFood:
      - name: "Kaffee, mit Koffein und Zucker"
        topLevelRecords:
        - type: "FOOD"
          name: "Kaffee, mit Koffein {assocRecp=492}"
          sid: "01339"
          facetSids: "0699,0201,1200,0701"
          amountConsumed: 150
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Zucker, Kristall"
          sid: "00992"
          facetSids: "1200"
          amountConsumed: 7.5
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
    - hourOfDay: "12:00:00"
      foodConsumptionOccasionId: "05"
      foodConsumptionPlaceId: "01"
      memorizedFood:
      - name: "Salat, gemischt mit Essig/Öl-Dressing"
        topLevelRecords:
        - type: "FOOD"
          name: "Salat, Eichblatt"
          sid: "00036"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salat, Kopf n.s."
          sid: "00040"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salat, Romana"
          sid: "00031"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salatdressing, Essig/Öl {assocRecp=384}"
          sid: "01527"
          facetSids: "0699,0499,1009,1200"
          amountConsumed: 12
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
      - name: "Salat, gemischt mit Joghurtdressing (Rezept bekannt)"
        topLevelRecords:
        - type: "FOOD"
          name: "Salat, Eichblatt"
          sid: "00036"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salat, Kopf n.s."
          sid: "00040"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salat, Romana"
          sid: "00031"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salatdressing, Jogurt {assocRecp=334}"
          sid: "01525"
          facetSids: "0699,0201,0499,1001,1501,1601"
          amountConsumed: 12
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking:
            type: "TYPE_OF_FAT_USED"
            name: "Öl, Raps"
            sid: "00937"
            facetSids: "1200"
          typeOfMilkOrLiquidUsedDuringCooking:
            type: "TYPE_OF_MILK_OR_LIQUID_USED"
            name: "Jogurt"
            sid: "00236"
            facetSids: "0102,0699,1200,0701,0801"
      - name: "Salat, gemisch mit Joghurtdressing (zubereitet in Rest.)"
        topLevelRecords:
        - type: "FOOD"
          name: "Salat, Eichblatt"
          sid: "00036"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salat, Kopf n.s."
          sid: "00040"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salat, Romana"
          sid: "00031"
          facetSids: "0204,0499,0910,2807"
          amountConsumed: 20
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Salatdressing, Jogurt {assocRecp=334}"
          sid: "01525"
          facetSids: "0699,0201,0400,1009,1200"
          amountConsumed: 12
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
      - name: "Pommes frites mit Ketchup"
        topLevelRecords:
        - type: "FOOD"
          name: "Pommes frites"
          sid: "00010"
          facetSids: "0204,0499,1200,0801,0310"
          amountConsumed: 142
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 25.560001373291016
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
        - type: "FOOD"
          name: "Ketchup"
          sid: "01515"
          facetSids: "0699,1200"
          amountConsumed: 28.399999618530273
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Pommes frites"
          sid: "00010"
          facetSids: "0204,0402,1200,0801,0309"
          amountConsumed: 142
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 11.359999656677246
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
        - type: "FOOD"
          name: "Pommes frites"
          sid: "00010"
          facetSids: "0204,0402,1200,0801,0307"
          amountConsumed: 142
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 5.679999828338623
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
        - type: "FOOD"
          name: "Pommes frites"
          sid: "00010"
          facetSids: "0204,0402,1200,0801,0309"
          amountConsumed: 142
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 11.359999656677246
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
        - type: "FOOD"
          name: "Pommes frites"
          sid: "00010"
          facetSids: "0204,0402,1200,0801,0301"
          amountConsumed: 142
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
    - hourOfDay: "14:00:00"
      foodConsumptionOccasionId: "07"
      foodConsumptionPlaceId: "01"
      memorizedFood:
      - name: "Kuchen"
        topLevelRecords:
        - type: "FOOD"
          name: "Kuchen n.s. {assocRecp=638}"
          sid: "01151"
          facetSids: "0699,1001,1500,1602,0702"
          amountConsumed: 80
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Kuchen n.s. {assocRecp=638}"
          sid: "01151"
          facetSids: "0699,1006,0702,0801"
          amountConsumed: 80
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Kuchen n.s. {assocRecp=638}"
          sid: "01151"
          facetSids: "0699,1009,0702"
          amountConsumed: 80
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
      - name: "Topfentorte"
        topLevelRecords:
        - type: "FOOD"
          name: "Topfentorte {assocRecp=434}"
          sid: "01179"
          facetSids: "0699,1001,1500,1602,0702"
          amountConsumed: 88
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FOOD"
          name: "Topfentorte {assocRecp=434}"
          sid: "01179"
          facetSids: "0699,1006,0702,0801"
          amountConsumed: 88
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
    - hourOfDay: "19:00:00"
      foodConsumptionOccasionId: "09"
      foodConsumptionPlaceId: "01"
      memorizedFood:
      - name: "Fisch, Lachs"
        topLevelRecords:
        - type: "FOOD"
          name: "Fisch, Lachs"
          sid: "00872"
          facetSids: "0131,0204,0499,1001,0328,1301,2807"
          amountConsumed: 56.549999237060547
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 1.6964999437332153
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
        - type: "FOOD"
          name: "Fisch, Lachs"
          sid: "00872"
          facetSids: "0131,0204,0402,1009,0905,0308,1301,2807"
          amountConsumed: 56.549999237060547
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 2.8275001049041748
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
      - name: "Gemüse"
        topLevelRecords:
        - type: "FOOD"
          name: "Gemüsemischung n.s."
          sid: "00017"
          facetSids: "0204,0499,0910,1200,0307,1302,2807"
          amountConsumed: 79
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
          typeOfFatUsedDuringCooking: null
          typeOfMilkOrLiquidUsedDuringCooking: null
        - type: "FRYING_FAT"
          name: "Öl, Raps"
          sid: "00937"
          facetSids: "1200"
          amountConsumed: 6.3199996948242188
          consumptionUnit: "GRAM"
          rawPerCookedRatio: 1
